### ISSUE TITLE
Share Link not working

### ISSUE THREAD

**Issue by DocKDE** [nextcloud/server#18502]
Hello,
I recently upgraded from 16.0.4 to 17.0.1. I just noticed that now the option to create a share link for a file or folder seems not to work anymore. The Button is there, it simply doesn't do anything. Could anyone help with that?
If additional info is required I can provide it.

### Steps to reproduce
1. Update from 16.0.4 to 17.0.1
2. Choose file or folder
3. Try to click on "Share Link"

### Expected behaviour
"Share Link" should produce a share link as before.

### Actual behaviour
Button does nothing, browser does not recognize it as a button

### Server configuration

**Operating system:** Debian 10 Buster (Rasbian)

**Web server:** Apache

**Database:** MariaDB

**PHP version:** 7.3

**Nextcloud version:** 17.0.1

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated from 16.0.4

**Where did you install Nextcloud from:** Tarball from website. Update was done first via Browser interface, after that didn't work I tried command line and that worked.

**Signing status:**
<details>
<summary>No errors have been found.</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
</details>

**List of activated apps:**
<details>
<summary>  

  - accessibility: 1.3.0
  - activity: 2.10.1
  - announcementcenter: 3.6.1
  - bruteforcesettings: 1.4.0
  - calendar: 1.7.1
  - cloud_federation_api: 1.0.0
  - comments: 1.7.0
  - contacts: 3.1.6
  - dav: 1.13.0
  - federatedfilesharing: 1.7.0
  - federation: 1.7.0
  - files: 1.12.0
  - files_external: 1.8.0
  - files_pdfviewer: 1.6.0
  - files_rightclick: 0.15.1
  - files_sharing: 1.9.0
  - files_trashbin: 1.7.0
  - files_versions: 1.10.0
  - files_videoplayer: 1.6.0
  - firstrunwizard: 2.6.0
  - gallery: 18.4.0
  - groupfolders: 5.0.4
  - logreader: 2.2.0
  - lookup_server_connector: 1.5.0
  - mail: 0.21.0
  - music: 0.11.1
  - news: 14.0.2
  - nextcloud_announcements: 1.6.0
  - notifications: 2.5.0
  - oauth2: 1.5.0
  - ojsxc: 3.4.4
  - password_policy: 1.7.0
  - privacy: 1.1.0
  - provisioning_api: 1.7.0
  - recommendations: 0.5.0
  - serverinfo: 1.7.0
  - sharebymail: 1.7.0
  - spreed: 7.0.2
  - support: 1.0.1
  - survey_client: 1.5.0
  - systemtags: 1.7.0
  - text: 1.1.1
  - theming: 1.8.0
  - twofactor_backupcodes: 1.6.0
  - updatenotification: 1.7.0
  - user_usage_report: 1.1.2
  - viewer: 1.2.0
  - workflowengine: 1.7.0
Disabled:
  - admin_audit
  - encryption
  - user_ldap

</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**Nextcloud configuration:**
<details>
<summary>

{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "192.168.178.48",
            "***REMOVED SENSITIVE VALUE***""
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "17.0.1.1",
        "overwrite.cli.url": "http:\/\/192.168.178.48\/nextcloud",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "maintenance": false,
        "has_rebuilt_cache": true,
        "theme": "",
        "loglevel": 2,
        "data-fingerprint": "cdc22d1de2b23fe6b7a91647f80e2a85",
        "app_install_overwrite": [
            "ojsxc"
        ]
    }
}

</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** I attached a HDD and moved the data folder there.

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** No

### Client configuration
**Browser:** Firefox

**Operating system:** Arch

#### Web server error log
<details>
<summary>[http2:warn] [pid648] AH10034: The mpm module (prefork.c) is not supported by mod_http2. The mpm determines how things are processed in your server. HTTP/2 has more demands in this regard and the currently selected mpm will just not do. This is an advisory warning. Your sever will continue to work but the HTTP/2 protocol will be inactive.

Basically there's nothing else here
</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>
Can't figure out how to get my hands on this at the moment.
</summary>

```
Insert your Nextcloud log here
```
</details>

**Comment by kesselb** [comment]
Thank you for reporting this issue :+1: 

Would you mind to edit your post and add the [issue template](https://github.com/nextcloud/server/blob/master/.github/ISSUE_TEMPLATE/Bug_report.md)? The information are required to reproduce your issue.

**Comment by DocKDE** [comment]
Done. Hope this helps some.

**Comment by kesselb** [comment]
I will need your browser console log to investigate this issue.  Open your console, reload your page and/or do the action leading to this issue and copy/paste the log in this thread.

<details>
<summary>How to access your browser console (Click to expand)</summary>

# Chrome
- Press either CTRL + SHIFT + J to open the “console” tab of the Developer Tools.
- Alternative method:
    1. Press either CTRL + SHIFT + I or F12 to open the Developer Tools.
    2. Click the “console” tab.

# Safari
- Press CMD + ALT + I to open the Web Inspector.
- See Chrome’s step 2. (Chrome and Safari have pretty much identical dev tools.)

# IE9
1. Press F12 to open the developer tools.
2. Click the “console” tab.

# Firefox
- Press CTRL + SHIFT + K to open the Web console (COMMAND + SHIFT + K on Macs).
- or, if Firebug is installed (recommended):
    1. Press F12 to open Firebug.
    2. Click on the “console” tab.

# Opera
1. Press CTRL + SHIFT + I to open Dragonfly.
2. Click on the “console” tab.
</details>

**Comment by DocKDE** [comment]
Sure, here you go. Sorry, this pulling of logs is still new to me. I found some messages of blocked content in there and I was able to determine that my uBlock Origin extension actually seems to block the plus-sign that I need to click on for sharing of links. This seems really odd to me because I've had this extension for a long time and never had this issue earlier. Is it possible that something changed on Nextcloud's side or is this probably due to some change in uBlock?

```
Content Security Policy: Directive ‘child-src’ has been deprecated. Please use directive ‘worker-src’ to control workers, or directive ‘frame-src’ to control frames respectively. 2
Content Security Policy: Ignoring duplicate source 'unsafe-inline'
Content Security Policy: Couldn’t parse invalid host 'unsafe-inline'
Content Security Policy: Directive ‘child-src’ has been deprecated. Please use directive ‘worker-src’ to control workers, or directive ‘frame-src’ to control frames respectively.
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). 2 utils.js:35:9
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). 3 content-script.js:40:65
oc_config is deprecated: use OC.config instead globals.js:26:15
This page uses the non standard property “zoom”. Consider using calc() in the relevant property values, or using “transform” along with “transform-origin: 0 0”. injectStylesIntoStyleTag.js:201:10
oc_requesttoken is deprecated: use OC.requestToken instead 2 globals.js:26:15
Proxying an event bus of version 0.2.1 with 0.2.0 ProxyBus.js:30:14
OCA.Viewer initialized Viewer.js:37:10
oc_config is deprecated: use OC.config instead globals.js:26:15
oc_appswebroots is deprecated: use OC.appswebroots instead globals.js:26:15
oc_config is deprecated: use OC.config instead globals.js:26:15
oc_appswebroots is deprecated: use OC.appswebroots instead globals.js:26:15
New search handler registered search.js:67:12
The escapeHTML library is deprecated! It will be removed in nextcloud 19. 2 globals.js:26:15
The select2 library is deprecated! It will be removed in nextcloud 19. globals.js:26:15
The select2 library is deprecated! It will be removed in nextcloud 19. globals.js:26:15
Content Security Policy: The report URI (about:blank) should be an HTTP or HTTPS URI.
Content Security Policy: The page’s settings observed the loading of a resource at https://thedarkside.spdns.de/nextcloud/apps/ojsxc/js/jsxc/lib/jsxc.otr.webworker.js (“worker-src”). A CSP report is being sent.
querying Whats New data was successful: nocontent whatsnew.js:42:9
undefined whatsnew.js:43:9
The humanFileSize library is deprecated! It will be removed in nextcloud 19. 2 globals.js:26:15
The humanFileSize library is deprecated! It will be removed in nextcloud 19. 4 globals.js:26:15
The escapeHTML library is deprecated! It will be removed in nextcloud 19. 3 globals.js:26:15
XML Parsing Error: not well-formed
Location: 
Line Number 1, Column 120: 11
XML Parsing Error: not well-formed
Location: 
Line Number 1, Column 120: 20
The select2 library is deprecated! It will be removed in nextcloud 19. 2 globals.js:26:15
The humanFileSize library is deprecated! It will be removed in nextcloud 19. 10 globals.js:26:15
The select2 library is deprecated! It will be removed in nextcloud 19. 2 globals.js:26:15
Content Security Policy: Directive ‘child-src’ has been deprecated. Please use directive ‘worker-src’ to control workers, or directive ‘frame-src’ to control frames respectively. 3
The escapeHTML library is deprecated! It will be removed in nextcloud 19. 10 globals.js:26:15
An iframe which has both allow-scripts and allow-same-origin for its sandbox attribute can remove its sandboxing. files
Content Security Policy: Ignoring ‘x-frame-options’ because of ‘frame-ancestors’ directive.
Content Security Policy: Ignoring duplicate source 'unsafe-inline'
Content Security Policy: Couldn’t parse invalid host 'unsafe-inline'
Content Security Policy: Directive ‘child-src’ has been deprecated. Please use directive ‘worker-src’ to control workers, or directive ‘frame-src’ to control frames respectively.
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). 2 utils.js:35:9
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). 3 content-script.js:40:65
The select2 library is deprecated! It will be removed in nextcloud 19. globals.js:26:15
The escapeHTML library is deprecated! It will be removed in nextcloud 19. 2 globals.js:26:15
The humanFileSize library is deprecated! It will be removed in nextcloud 19. globals.js:26:15
Content Security Policy: Directive ‘child-src’ has been deprecated. Please use directive ‘worker-src’ to control workers, or directive ‘frame-src’ to control frames respectively. 3
An iframe which has both allow-scripts and allow-same-origin for its sandbox attribute can remove its sandboxing. files
Content Security Policy: Ignoring ‘x-frame-options’ because of ‘frame-ancestors’ directive.
Content Security Policy: Ignoring duplicate source 'unsafe-inline'
Content Security Policy: Couldn’t parse invalid host 'unsafe-inline'
Content Security Policy: Directive ‘child-src’ has been deprecated. Please use directive ‘worker-src’ to control workers, or directive ‘frame-src’ to control frames respectively.
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). 2 utils.js:35:9
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). 2 content-script.js:40:65
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). content-script.js:40:65
#Close is undefined. l10n.js:831:14
#Close is undefined. l10n.js:903:14
#close_label is undefined. l10n.js:831:14
#close_label is undefined. l10n.js:903:14
Content Security Policy: The report URI (about:blank) should be an HTTP or HTTPS URI.
Content Security Policy: The page’s settings observed the loading of a resource at https://thedarkside.spdns.de/nextcloud/apps/files_pdfviewer/vendor/pdfjs/build/pdf.worker.js?v=1.6.0 (“worker-src”). A CSP report is being sent.
```

**Comment by kesselb** [comment]
Thanks :+1: 

There is not much we can do. It's probably fixed with Nextcloud 18. Check https://github.com/nextcloud/server/issues/14616 for more details. Okay with you to close this one?

**Comment by DocKDE** [comment]
Yeah sure, close it up. Just glad the problem was a simple fix.
